hello-world/hello_world_test.rb

```diff
@@ -12,33 +12,25 @@
   exit 1
 end
 
+# Test data version:
+# 7668b09 Added hello world test definition
+
 class HelloWorldTest < Minitest::Test
   def test_no_name
-    assert_equal 'Hello, World!', HelloWorld.hello, <<-MSG
-    When given no name, it should greet the world!
-    MSG
+    assert_equal 'Hello, World!', HelloWorld.hello,
+                 'When given no name, it should greet the world!'
   end
```

Comment:
I agree, they are all box ticking, and generally the messages aren't needed if the tests are well named.

I think the best is to ditch messages, and if they are really needed, then use a heretdoc as they probably need to be multiple lines then anyway.

Happy with this approach?


> On 24 Oct 2015, at 21:01, Katrina Owen <notifications@github.com> wrote:
> 
> In hello-world/hello_world_test.rb:
> 
> >  class HelloWorldTest < Minitest::Test
> >    def test_no_name
> > -    assert_equal 'Hello, World!', HelloWorld.hello, <<-MSG
> > -    When given no name, it should greet the world!
> > -    MSG
> > +    assert_equal 'Hello, World!', HelloWorld.hello,
> > +                 'When given no name, it should greet the world!'
> >    end
> Thanks! I completely missed this one last time around.
> 
> You've listed several options:
> 
> assign a temporary variable
> use a heredoc
> break the line
> make the message shorter
> Every single one of these are a box-ticking exercise, because Rubocop (and the Ruby community in general) has decided that 80 characters is the line limit.
> 
> There's an additional option:
> 
> just ditch using extra messages
> Every once in a while the assertion failure doesn't have good inherent understandability, and an extra message is helpful (and a good practice). The rest of the time messages are noise. They add to the code, they add to the error message, and if they're not pulling their weight we should just ditch them.
> 
> The question is: is it worth having support for them to handle the few cases when they're useful?
> 
> If I were to rank these options in order of I hate it the most to I hate it the least, my list would look like this:
> 
> break the line (I triple hate this with fudge on top)
> make the message shorter (defeats the purpose)
> use a heredoc (bulky, more code)
> assign a tmp (yeah, whatevs)
> just ditch messages altogether (seriously, not many people use them)
> I think I've changed my mind a bit on messages overall over the past couple of years. I think that people should know how to do it, and understand how to read the assertions / test suite when they're there, and I used to think it was important enough to add this to exercism, but honestly it's not worth the headache of ticking rubocop boxes.
> 
> What do you think? tmp var or ditch them altogether?
> 
> —
> Reply to this email directly or view it on GitHub.
>